src/util/string.h

```diff
@@ -761,3 +761,26 @@ std::string sanitizeDirName(const std::string &str, const std::string &optional_
  * brackets (e.g. "a\x1eb" -> "a<1e>b").
  */
 void safe_print_string(std::ostream &os, const std::string &str);
+
+
+// Tell the compiler that a function has printf-style arguments,
+// so that it can catch safety errors at compile-time.
+#ifdef _MSC_VER
+#define PRINTF_ATTR(index_of_format, index_of_first_vararg)
+#define PRINTF_FORMAT_PREFIX _Printf_format_string_
+#else
+#define PRINTF_ATTR(index_of_format, index_of_first_vararg) \
+	__attribute__ ((format (printf, index_of_format, index_of_first_vararg)))
+#define PRINTF_FORMAT_PREFIX
+#endif
+
+/**
+ *
+ * StringPrintf
+ *
+ * Create a std::string using a format string like printf.
+ *
+ */
+
+std::string StringPrintf(PRINTF_FORMAT_PREFIX const char *format, ...) PRINTF_ATTR(1, 2);
```

Comment:
should probably be called `string_printf` or maybe `strprintf`

Reply:
`StringPrintf` is the name used in chromium (and android) for the same function. I would prefer this just out of habit, but if it is too inconsistent, I could replace it with `strprintf` (less typing anyway).